### PR TITLE
ci: Update CODEOWNERS file

### LINKED ISSUE #456: Add Repo Maintainers to CODEOWNERS file
# Update CODEOWNERS to provide maintainers ownership of all files in their repos

**Description**:

Update this repositories `.github/CODEOWNER` file to ensure that the repository maintainer
group(s) have full control of the files in this repository.

### PR DESCRIPTION
**Pull request by mishomihov00**
**Description**:

Updates the CODEOWNERS file

**Related issue(s)**:

Fixes #456 

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

**Review by san-est — APPROVED** [review]
LGTM

**Review by andrewb1269hg — DISMISSED** [review]

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -1,28 +1,33 @@
...
 #####  Core Files  ######
 #########################
 
 # NOTE: Must be placed last to ensure enforcement over all other rules
 
 # Protection Rules for Github Configuration Files and Actions Workflows
-/.github/                                       @hashgraph/release-engineering-managers @hashgraph/devops-ci @hashgraph/devops-ci-committers
-/.github/workflows/                             @hashgraph/devops-ci @hashgraph/devops-ci-committers
+/.github/                                       @hashgraph/release-engineering-managers @hashgraph/devops-ci
+/.github/workflows/                             @hashgraph/devops-ci
```
Comment (rbarkerSL) [inline .github/CODEOWNERS:19]:
```suggestion
/.github/workflows/                             @hashgraph/devops-ci @hashgraph/devops-ci-committers @hashgraph/release-engineering-managers 
```

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -1,28 +1,33 @@
...
 
 # Self-protection for root CODEOWNERS files (this file should not exist and should definitely require approval)
 /CODEOWNERS                                     @hashgraph/release-engineering-managers
 
 # Protect the repository root files
-/README.md                                      @hashgraph/release-engineering-managers @hashgraph/devops-ci @hashgraph/devops-ci-committers
+/README.md                                      @hashgraph/release-engineering-managers @hashgraph/devops-ci
```
Comment (rbarkerSL) [inline .github/CODEOWNERS:25]:
```suggestion
/README.md                                      @hashgraph/devops-ci @hashgraph/release-engineering-managers @hashgraph/developer-advocates @hashgraph/platform-base @hashgraph/hedera-services @hashgraph/mirror-node @hashgraph/hedera-smart-contracts @hashgraph/block-node
```

File: .github/CODEOWNERS [hunk 3]
```diff
@@ -1,28 +1,33 @@
...
 
 # CodeCov configuration
-**/codecov.yml                                  @hashgraph/release-engineering-managers @hashgraph/devops-ci @hashgraph/devops-ci-committers
+**/codecov.yml                                  @hashgraph/release-engineering-managers @hashgraph/devops-ci
 
 # Git Ignore definitions
-**/.gitignore                                   @hashgraph/release-engineering-managers @hashgraph/devops-ci @hashgraph/devops-ci-committers
-**/.gitignore.*                                 @hashgraph/release-engineering-managers @hashgraph/devops-ci @hashgraph/devops-ci-committers
+**/.gitignore                                   @hashgraph/release-engineering-managers @hashgraph/devops-ci
+**/.gitignore.*                                 @hashgraph/release-engineering-managers @hashgraph/devops-ci
```
Comment (rbarkerSL) [inline .github/CODEOWNERS:33]:
```suggestion
**/.gitignore                                   @hashgraph/devops-ci @hashgraph/release-engineering-managers @hashgraph/developer-advocates @hashgraph/platform-base @hashgraph/hedera-services @hashgraph/mirror-node @hashgraph/hedera-smart-contracts @hashgraph/block-node
**/.gitignore.*                                 @hashgraph/devops-ci @hashgraph/release-engineering-managers @hashgraph/developer-advocates @hashgraph/platform-base @hashgraph/hedera-services @hashgraph/mirror-node @hashgraph/hedera-smart-contracts @hashgraph/block-node
```

**Review by rbarkerSL — CHANGES_REQUESTED** [review]

File: .github/CODEOWNERS [hunk 4]
```diff
@@ -1,28 +1,33 @@
-*                                               @hashgraph/hedera-services @hashgraph/mirror-node
-/platform/                                      @hashgraph/platform-hashgraph
+###################################
+##### Global Protection Rule ######
+###################################
+# NOTE: This rule is overriden by the more specific rules below. This is the catch-all rule for all files not covered by the more specific rules below
+*                                               @hashgraph/devops-ci @hashgraph/release-engineering-managers @hashgraph/developer-advocates @hashgraph/platform-base @hashgraph/hedera-services @hashgraph/mirror-node @hashgraph/hedera-smart-contracts @hashgraph/block-node
+
+#Platform and mirror folder rules
```
Comment (jsync-swirlds) [inline .github/CODEOWNERS:7]: We might consider adding the following specific rules:
```
# Block Node Folder Rules
/block/                                           @hashgraph/block-node @hashgraph/hedera-services @hashgraph/mirror-node
/services/                                        @hashgraph/hedera-services @hashgraph/mirror-node
```

**Review by andrewb1269hg — APPROVED** [review]

**Review by jsync-swirlds — APPROVED** [review]

**Review by rbarkerSL — APPROVED** [review]

**Pull request merged**